index.bs

```diff
@@ -872,88 +872,93 @@ When this method is invoked, the user agent MUST execute the following algorithm
-    devices are hot-plugged into (USB) or discovered by (NFC) browsers, and are underspecified.
-    Resolving this with good definitions or some other means will be addressed by resolving
-    [Issue #613](https://github.com/w3c/webauthn/issues/613).
+    Note: What qualifies an [=authenticator=] as "available" is intentionally unspecified; this is meant to represent how
+    [=authenticators=] can be hot-plugged into (e.g., via USB) or discovered (e.g., via NFC or Bluetooth) by the [=client=] by
+    various mechanisms, or permanently built into the [=client=].
+
+1. Start |lifetimeTimer|.
 
-    1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}</code> is [=present=]:
+1. [=While=] |lifetimeTimer| has not expired, perform the following actions depending upon |lifetimeTimer|,
+    and the state and response [=set/for each=] |authenticator| in |authenticators|:
+    <dl class="switch">
+        :   If |lifetimeTimer| expires,
+        ::  [=set/For each=] |authenticator| in |issuedRequests| invoke the [=authenticatorCancel=] operation on |authenticator|
+            and [=set/remove=] |authenticator| from |issuedRequests|.
 
-          1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{authenticatorAttachment}}</code> is
-            [=present|present=] and its value is not equal to |authenticator|'s attachment modality, [=iteration/continue=].
-          1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{requireResidentKey}}</code> is set to
-            [TRUE] and the |authenticator| is not capable of storing a [=Client-Side-Resident Credential Private Key=],
-            [=iteration/continue=].
-          1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{AuthenticatorSelectionCriteria/userVerification}}</code> is
-            set to {{UserVerificationRequirement/required}} and the |authenticator| is not capable of performing [=user
-            verification=], [=iteration/continue=].
+        :   If the <code>|options|.{{CredentialCreationOptions/signal}}</code> is [=present=] and its
+            [=AbortSignal/aborted flag=] is set to [TRUE],
+        ::  [=set/For each=] |authenticator| in |issuedRequests| invoke the [=authenticatorCancel=]
+            operation on |authenticator| and [=set/remove=] |authenticator| from |issuedRequests|. Then return a {{DOMException}}
+            whose name is "{{AbortError}}" and terminate this algorithm.
 
-    1. Let |userVerification| be the <dfn>effective user verification requirement for credential creation</dfn>, a Boolean value,
-        as follows. If
-        <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{AuthenticatorSelectionCriteria/userVerification}}</code>
+        :   If an |authenticator| becomes available on this platform,
```

Comment:
I was considering something like that, to emphasize authenticators that are already available, but I thought it might be confusing since this is all wrapped in a "while" loop. That would technically require something like "is, or becomes, available on this platform and is not present in |issuedRequests|" to prevent issuing an unlimited number of requests to it, but then it would still trigger again after another of the switch cases causes it to be removed from |issuedRequests|.

I guess it's probably obvious enough what we would mean by that, so I'm ready to agree if you're positive it would be an improvement. It just makes my inner mathematician cringe a little bit. :)

Reply:
yeah, hm, let's leave it as-is, thx for cogent explanation.